_overviews/core/custom-collection-operations.md

```diff
@@ -259,13 +400,38 @@ class IntersperseOperation[A](seqOps: SeqOps[A, Iterable, _]) {
 implicit def IntersperseOperation[Repr](coll: Repr)(implicit seq: IsSeq[Repr]): IntersperseOperation[seq.A] =
   new IntersperseOperation(seq(coll))
 ~~~
+{% endtab %}
+{% tab 'Scala 3' for=intersperse_2 %}
+~~~ scala
+import scala.collection.{ AbstractIterator, AbstractView, Factory, SeqOps }
+import scala.collection.generic.IsSeq
+
+extension[A](seqOps: SeqOps[A, Iterable, _])
```

Comment:
This whole code example should be replaced by the following, which uses `IsSeq`:
```scala
import scala.collection.{ AbstractIterator, AbstractView, Factory, SeqOps }
import scala.collection.generic.IsSeq

extension [Repr](coll: Repr)(using seq: IsSeq[Repr])
  def intersperse[B >: seq.A, That](sep: B)(using factory: Factory[B, That]): That =
    val seqOps = seq(coll)
    factory.fromSpecific(new AbstractView[B]:
      def iterator = new AbstractIterator[B]:
        val it = seqOps.iterator
        var intersperseNext = false
        def hasNext = intersperseNext || it.hasNext
        def next() =
          val elem = if intersperseNext then sep else it.next()
          intersperseNext = !intersperseNext && it.hasNext
          elem
    )
```

Reply:
I agree. I propose to make the same update for scala 2 as well... I will make a commit with these updates